account_payment_return/models/payment_return_reason.py

```diff
@@ -11,9 +12,22 @@ class PaymentReturnReason(models.Model):
 
     code = fields.Char()
     name = fields.Char(string="Reason", translate=True)
-    display_name = fields.Char(compute="_compute_display_name")
+    display_name = fields.Char(
+        compute="_compute_display_name", search="_search_display_name"
+    )
 
     @api.depends("code", "name")
     def _compute_display_name(self):
         for record in self:
             record.display_name = f"[{record.code}] {record.name}"
+
+    @api.model
+    def _search_display_name(self, operator, value):
+        domain = []
+        for f_name in self._rec_names_search:
+            domain_item = [(f_name, operator, value)]
+            if domain == []:
+                domain = domain_item
+            else:
+                domain = expression.OR([domain, domain_item])
```

Comment:
The join should be `AND` if the operator is in negative operator list (there's a constant list that you can import to check its inclusion).